lib/ansible/plugins/action/__init__.py

```diff
@@ -291,28 +291,49 @@ def _remote_chmod(self, mode, path, sudoable=False):
         res = self._low_level_execute_command(cmd, sudoable=sudoable)
         return res
 
-    def _remote_checksum(self, path, all_vars):
+    def _execute_remote_stat(self, path, all_vars, follow):
         '''
-        Takes a remote checksum and returns 1 if no file
+        Get information from remote file.
         '''
+        module_args=dict(
+           path=path,
+           follow=follow,
+           get_md5=False,
+           get_checksum=True,
+           checksum_algo='sha1',
+        )
+        mystat = self._execute_module(module_name='stat', module_args=module_args, task_vars=all_vars)
+
+        if 'failed' in mystat and mystat['failed']:
+            raise AnsibleError('Failed to get information on remote file (%s): %s' % (path, mystat['msg']))
 
-        python_interp = all_vars.get('ansible_python_interpreter', 'python')
+        if not mystat['stat']['exists']:
+            # empty might be matched, 1 should never match, also backwards compatible
+            mystat['stat']['checksum'] = '1'
 
-        cmd = self._connection._shell.checksum(path, python_interp)
-        data = self._low_level_execute_command(cmd, sudoable=True)
+        return mystat['stat']
+
+    def _remote_checksum(self, path, all_vars):
+        '''
+        Takes a remote checksum and returns 1 if no file
+        Returns a number 0-4 for specific errors.
+        '''
```

Comment:
Minor: documentation should be something like:
```
Takes a remote filename and returns 1 if no file.  Returns a number 0-4 for specific errors.  Returns the checksum (currently sha1) of the file otherwise.
```